### DISCUSSION TITLE
Flux  Image updation with flagger rollback

### DISCUSSION THREAD

**Discussion by rverma-dev** [fluxcd/flux2#1822] [Q&A]
Currently we have flux image updation, a new release of image tag will cause a git commit with the image tag. When flux start reconciling, flagger takes over and do a blue green deployment. Sometime the deployment failed and the actual image wasn't actually released.

In that scenario there is a mismatch in what the state is reflected in git and what is actual at cluster.
Can we use the ImageAutomation to add a comment of image tag in cluster too? 
So that within the git we know the actual state? I am open to other suggestions too :)

**Comment by kingdonb** [comment]
The comments are not persisted into the cluster, so that will not be possible.

I think you want to have an alert that fires when a flagger canary fails, since this is an alert-level event, something has gone wrong enough with a deployment to cause a rollback, and now some intervention is possibly needed as you've pointed to at least reconcile the git state with reality.

There is a section of the Flagger docs about alerting: https://docs.flagger.app/usage/alerting#prometheus-alert-manager

You can use Slack or even Prometheus AlertManager to handle this. I do not think it will be possible for Flagger to write back its failed status to the git commit any time soon.
